### PR TITLE
graphviz:  for stopping errors with --with-bindings 

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
----
This is the action for https://github.com/Homebrew/homebrew-core/issues/19110.

And I encountered another error with `php56`.
```
Last 15 lines from /Users/s172262/Library/Logs/Homebrew/graphviz/02.make:
  value = (swig_object_wrapper *) zend_list_find((*args[0])->value.lval, &type);
                                  ^~~~~~~~~~~~~~
                                  zend_hash_find
/usr/include/php/Zend/zend_hash.h:161:30: note: 'zend_hash_find' declared here
ZEND_API zval* ZEND_FASTCALL zend_hash_find(const HashTable *ht, zend_string *key);
                             ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
  CXXLD    libgv_java.la
20 errors generated.
make[4]: *** [libgv_php_la-gv_php.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [install-recursive] Error 1
make[2]: *** [install-recursive] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2
```
Php is out of scope in this formula. So I would add "--enable-php=no" option to avoid this error.

### CODE REVIEW

File: Formula/graphviz.rb [hunk 1]
```diff
@@ -48,6 +48,10 @@ class Graphviz < Formula
   end
 
   def install
+    # The following setting makes to prohibit using Homebrew's Python. See :
+    # https://github.com/Homebrew/homebrew-core/issues/19110
+    ENV["PYTHON"]="/usr/bin/python"
```
Comment (DomT4) [inline Formula/graphviz.rb]: The `python` flexibility is an intentional thing here, or the dependency wouldn't be `depends_on :python`.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
This should work fine with the stock `depends_on :python`, which deliberately favors Homebrew's python regardless of your PATH.

**Comment by ilovezfs** [comment]
Also the php bindings should be fixed as opposed to turning them off.

**Comment by s172262** [comment]
To turn `php` off is workaround of upstream.
See :http://www.graphviz.org/mantisbt/view.php?id=2584
>To build the rest of graphviz in the presence of php-7, you can use:
    ./configure --disable-php ....

**Comment by s172262** [comment]
Error of `gprof2dot` is caused from https://github.com/Homebrew/homebrew-core/commit/16ebe5f1843e6cb54856311ff0f676be53007329. `gprof2dot` needs to restore.
Testing  with `gprof2dot` of previous version is O.K.

Testing `gprof2dot` with current version:
```
$ brew test gprof2dot
Testing gprof2dot
==> /usr/local/Cellar/gprof2dot/2017.9.19/bin/gprof2dot /tmp/gprof2dot-test-2017
Last 15 lines from /Users/aoyamatakuo/Library/Logs/Homebrew/gprof2dot/test.01.gprof2dot:
2017-10-21 11:22:22 +0900

/usr/local/Cellar/gprof2dot/2017.9.19/bin/gprof2dot
/tmp/gprof2dot-test-20171021-50288-e0rml5/gprof.output
-o
/tmp/gprof2dot-test-20171021-50288-e0rml5/call_graph.dot

error: unexpected end of file
Error: gprof2dot: failed
```
"gprof.output" made by current `gprof2dot`:
```
    Flat profile:

    Each sample counts as 0.01 seconds.
     no time accumulated

      %   cumulative   self              self     total
     time   seconds   seconds    calls  Ts/call  Ts/call  name
      0.00      0.00     0.00        2     0.00     0.00  manager
      0.00      0.00     0.00        2     0.00     0.00  worker1
      0.00      0.00     0.00        2     0.00     0.00  worker2
      0.00      0.00     0.00        1     0.00     0.00  project1
      0.00      0.00     0.00        1     0.00     0.00  project2

     %         the percentage of the total running time of the
    time       program used by this function.

    cumulative a running sum of the number of seconds accounted
     seconds   for by this function and those listed above it.

     self      the number of seconds accounted for by this
    seconds    function alone.  This is the major sort for this
               listing.

    calls      the number of times this function was invoked, if
               this function is profiled, else blank.

     self      the average number of milliseconds spent in this
    ms/call    function per call, if this function is profiled,
           else blank.

     total     the average number of milliseconds spent in this
    ms/call    function and its descendents per call, if this
           function is profiled, else blank.

    name       the name of the function.  This is the minor sort
               for this listing. The index shows the location of
           the function in the gprof listing. If the index is
           in parenthesis it shows where it would appear in
           the gprof listing if it were to be printed.

                 Call graph (explanation follows)


    granularity: each sample hit covers 2 byte(s) no time propagated

    index % time    self  children    called     name
                    0.00    0.00       1/2           project1 [4]
                    0.00    0.00       1/2           project2 [5]
    [1]      0.0    0.00    0.00       2         manager [1]
                    0.00    0.00       2/2           worker1 [2]
                    0.00    0.00       2/2           worker2 [3]
    -----------------------------------------------
                    0.00    0.00       2/2           manager [1]
    [2]      0.0    0.00    0.00       2         worker1 [2]
    -----------------------------------------------
                    0.00    0.00       2/2           manager [1]
    [3]      0.0    0.00    0.00       2         worker2 [3]
    -----------------------------------------------
                    0.00    0.00       1/1           main [12]
    [4]      0.0    0.00    0.00       1         project1 [4]
                    0.00    0.00       1/2           manager [1]
    -----------------------------------------------
                    0.00    0.00       1/1           main [12]
    [5]      0.0    0.00    0.00       1         project2 [5]
                    0.00    0.00       1/2           manager [1]
    -----------------------------------------------

     This table describes the call tree of the program, and was sorted by
     the total amount of time spent in each function and its children.

     Each entry in this table consists of several lines.  The line with the
     index number at the left hand margin lists the current function.
     The lines above it list the functions that called this function,
     and the lines below it list the functions this one called.
     This line lists:
         index    A unique number given to each element of the table.
            Index numbers are sorted numerically.
            The index number is printed next to every function name so
            it is easier to look up where the function in the table.

         % time    This is the percentage of the `total' time that was spent
            in this function and its children.  Note that due to
            different viewpoints, functions excluded by options, etc,
            these numbers will NOT add up to 100%.

         self    This is the total amount of time spent in this function.

         children    This is the total amount of time propagated into this
            function by its children.

         called    This is the number of times the function was called.
            If the function called itself recursively, the number
            only includes non-recursive calls, and is followed by
            a `+' and the number of recursive calls.

         name    The name of the current function.  The index number is
            printed after it.  If the function is a member of a
            cycle, the cycle number is printed between the
            function's name and the index number.


     For the function's parents, the fields have the following meanings:

         self    This is the amount of time that was propagated directly
            from the function into this parent.

         children    This is the amount of time that was propagated from
            the function's children into this parent.

         called    This is the number of times this parent called the
            function `/' the total number of times the function
            was called.  Recursive calls to the function are not
            included in the number after the `/'.

         name    This is the name of the parent.  The parent's index
            number is printed after it.  If the parent is a
            member of a cycle, the cycle number is printed between
            the name and the index number.

     If the parents of the function cannot be determined, the word
     `<spontaneous>' is printed in the `name' field, and all the other
     fields are blank.

     For the function's children, the fields have the following meanings:

         self    This is the amount of time that was propagated directly
            from the child into the function.

         children    This is the amount of time that was propagated from the
            child's children to the function.

         called    This is the number of times the function called
            this child `/' the total number of times the child
            was called.  Recursive calls by the child are not
            listed in the number after the `/'.

         name    This is the name of the child.  The child's index
            number is printed after it.  If the child is a
            member of a cycle, the cycle number is printed
            between the name and the index number.

     If there are any cycles (circles) in the call graph, there is an
     entry for the cycle-as-a-whole.  This entry shows who called the
     cycle (as parents) and the members of the cycle (as children.)
     The `+' recursive calls entry shows the number of function calls that
     were internal to the cycle, and the calls entry for each member shows,
     for that member, how many times it was called from other members of
     the cycle.

    
    Index by function name

       [1] manager                 [5] project2                [3] worker2
       [4] project1                [2] worker1

```

"gprof.output" made by `gprof2dot` of previous version:
```
Flat profile:

Each sample counts as 0.01 seconds.
 no time accumulated

  %   cumulative   self              self     total
 time   seconds   seconds    calls  Ts/call  Ts/call  name
  0.00      0.00     0.00        2     0.00     0.00  manager
  0.00      0.00     0.00        2     0.00     0.00  worker1
  0.00      0.00     0.00        2     0.00     0.00  worker2
  0.00      0.00     0.00        1     0.00     0.00  project1
  0.00      0.00     0.00        1     0.00     0.00  project2

 %         the percentage of the total running time of the
time       program used by this function.

cumulative a running sum of the number of seconds accounted
 seconds   for by this function and those listed above it.

 self      the number of seconds accounted for by this
seconds    function alone.  This is the major sort for this
           listing.

calls      the number of times this function was invoked, if
           this function is profiled, else blank.

 self      the average number of milliseconds spent in this
ms/call    function per call, if this function is profiled,
       else blank.

 total     the average number of milliseconds spent in this
ms/call    function and its descendents per call, if this
       function is profiled, else blank.

name       the name of the function.  This is the minor sort
           for this listing. The index shows the location of
       the function in the gprof listing. If the index is
       in parenthesis it shows where it would appear in
       the gprof listing if it were to be printed.

             Call graph (explanation follows)


granularity: each sample hit covers 2 byte(s) no time propagated

index % time    self  children    called     name
                0.00    0.00       1/2           project1 [4]
                0.00    0.00       1/2           project2 [5]
[1]      0.0    0.00    0.00       2         manager [1]
                0.00    0.00       2/2           worker1 [2]
                0.00    0.00       2/2           worker2 [3]
-----------------------------------------------
                0.00    0.00       2/2           manager [1]
[2]      0.0    0.00    0.00       2         worker1 [2]
-----------------------------------------------
                0.00    0.00       2/2           manager [1]
[3]      0.0    0.00    0.00       2         worker2 [3]
-----------------------------------------------
                0.00    0.00       1/1           main [12]
[4]      0.0    0.00    0.00       1         project1 [4]
                0.00    0.00       1/2           manager [1]
-----------------------------------------------
                0.00    0.00       1/1           main [12]
[5]      0.0    0.00    0.00       1         project2 [5]
                0.00    0.00       1/2           manager [1]
-----------------------------------------------

 This table describes the call tree of the program, and was sorted by
 the total amount of time spent in each function and its children.

 Each entry in this table consists of several lines.  The line with the
 index number at the left hand margin lists the current function.
 The lines above it list the functions that called this function,
 and the lines below it list the functions this one called.
 This line lists:
     index    A unique number given to each element of the table.
        Index numbers are sorted numerically.
        The index number is printed next to every function name so
        it is easier to look up where the function in the table.

     % time    This is the percentage of the `total' time that was spent
        in this function and its children.  Note that due to
        different viewpoints, functions excluded by options, etc,
        these numbers will NOT add up to 100%.

     self    This is the total amount of time spent in this function.

     children    This is the total amount of time propagated into this
        function by its children.

     called    This is the number of times the function was called.
        If the function called itself recursively, the number
        only includes non-recursive calls, and is followed by
        a `+' and the number of recursive calls.

     name    The name of the current function.  The index number is
        printed after it.  If the function is a member of a
        cycle, the cycle number is printed between the
        function's name and the index number.


 For the function's parents, the fields have the following meanings:

     self    This is the amount of time that was propagated directly
        from the function into this parent.

     children    This is the amount of time that was propagated from
        the function's children into this parent.

     called    This is the number of times this parent called the
        function `/' the total number of times the function
        was called.  Recursive calls to the function are not
        included in the number after the `/'.

     name    This is the name of the parent.  The parent's index
        number is printed after it.  If the parent is a
        member of a cycle, the cycle number is printed between
        the name and the index number.

 If the parents of the function cannot be determined, the word
 `<spontaneous>' is printed in the `name' field, and all the other
 fields are blank.

 For the function's children, the fields have the following meanings:

     self    This is the amount of time that was propagated directly
        from the child into the function.

     children    This is the amount of time that was propagated from the
        child's children to the function.

     called    This is the number of times the function called
        this child `/' the total number of times the child
        was called.  Recursive calls by the child are not
        listed in the number after the `/'.

     name    This is the name of the child.  The child's index
        number is printed after it.  If the child is a
        member of a cycle, the cycle number is printed
        between the name and the index number.

 If there are any cycles (circles) in the call graph, there is an
 entry for the cycle-as-a-whole.  This entry shows who called the
 cycle (as parents) and the members of the cycle (as children.)
 The `+' recursive calls entry shows the number of function calls that
 were internal to the cycle, and the calls entry for each member shows,
 for that member, how many times it was called from other members of
 the cycle.


Index by function name

   [1] manager                 [5] project2                [3] worker2
   [4] project1                [2] worker1
```

**Comment by ilovezfs** [comment]
There's a form feed character on line 68 causing it.

**Comment by ilovezfs** [comment]
@s172262 fixed in https://github.com/Homebrew/homebrew-core/pull/19683.

**Comment by s172262** [comment]
@BrewTestBot test this please

**Comment by s172262** [comment]
Since Jenkins fails to handling this branch, I close this PR and open new one.

**Pull request closed without merging**